examples/LiftSim_baseline/A2C/train.py

```diff
@@ -0,0 +1,226 @@
+# limitations under the License.
+
+import numpy as np
+import os
+import parl
+import queue
+import six
+import time
+import threading
+
+from actor import Actor
+from collections import defaultdict
+from env_wrapper import ObsProcessWrapper, ActionProcessWrapper
+from parl.utils import logger, get_gpu_count, tensorboard, machine_info
+from parl.utils.scheduler import PiecewiseScheduler
+from parl.utils.time_stat import TimeStat
+from parl.utils.window_stat import WindowStat
+from rlschool import LiftSim
+from lift_model import LiftModel
+from lift_agent import LiftAgent
+
+
+class Learner(object):
+    def __init__(self, config):
+        self.config = config
+
+        #=========== Create Agent ==========
+        env = LiftSim()
+        env = ActionProcessWrapper(env)
+        env = ObsProcessWrapper(env)
+
+        obs_dim = env.obs_dim
+        act_dim = env.act_dim
+        self.config['obs_dim'] = obs_dim
+
+        model = LiftModel(act_dim)
+        algorithm = parl.algorithms.A3C(
+            model, vf_loss_coeff=config['vf_loss_coeff'])
+        self.agent = LiftAgent(algorithm, config)
+
+        if machine_info.is_gpu_available():
+            assert get_gpu_count() == 1, 'Only support training in single GPU,\
+                    Please set environment variable: `export CUDA_VISIBLE_DEVICES=[GPU_ID_TO_USE]` .'
+
+        else:
+            cpu_num = os.environ.get('CPU_NUM')
+            assert cpu_num is not None and cpu_num == '1', 'Only support training in single CPU,\
+                    Please set environment variable:  `export CPU_NUM=1`.'
+
+        #========== Learner ==========
+
+        self.entropy_stat = WindowStat(100)
+        self.target_values = None
+
+        self.learn_time_stat = TimeStat(100)
+        self.start_time = None
+
+        #========== Remote Actor ===========
+        self.remote_count = 0
+        self.sample_data_queue = queue.Queue()
+
+        self.remote_metrics_queue = queue.Queue()
+        self.sample_total_steps = 0
+
+        self.params_queues = []
+        self.create_actors()
+
+        self.log_steps = 0
+
+    def create_actors(self):
+        """ Connect to the cluster and start sampling of the remote actor.
+        """
+        parl.connect(self.config['master_address'])
+
+        logger.info('Waiting for {} remote actors to connect.'.format(
+            self.config['actor_num']))
+
+        for i in six.moves.range(self.config['actor_num']):
+            params_queue = queue.Queue()
+            self.params_queues.append(params_queue)
+
+            self.remote_count += 1
+            logger.info('Remote actor count: {}'.format(self.remote_count))
+
+            remote_thread = threading.Thread(
+                target=self.run_remote_sample, args=(params_queue, ))
+            remote_thread.setDaemon(True)
+            remote_thread.start()
+
+        logger.info('All remote actors are ready, begin to learn.')
```

Comment:
When resource is scarce for actors, it is inappropriate to print this log.
I suggest deleting this log.

Reply:
deleted